result/tests.py

```diff
@@ -75,35 +75,45 @@ def test_no_arg_ok():
 
 
 def test_expect():
+    err_message = 'failure'
+
     o = Ok('yay')
     n = Err('nay')
-    assert o.expect('failure') == 'yay'
-    with pytest.raises(UnwrapError):
-        n.expect('failure')
+    assert o.expect(err_message) == 'yay'
+    with pytest.raises(UnwrapError) as e:
+        n.expect(err_message)
+    assert str(e.value) == err_message
+    assert e.value.result is n
```

Comment:
Ideally we'd put a single standalone test to assert the result is in the exception so we don't have to repeat it in multiple tests